### PR TITLE
Fix/11116 remove info button icon | 9640 Wrong text in UQS-Warn Others

### PR DESCRIPTION
**Pull request by puneetmahali1**
## Description
- (EXPOSUREAPP-11116): **Remove the Information Icon from "Warn for Others" QR Scanner screen**

- (EXPOSUREAPP-9640): **Change the text on "Warn for Others" QR Scanner screen**

## Link to Jira
- (EXPOSUREAPP-11116) - https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-11116
- (EXPOSUREAPP-9640) - https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-9640

## Screenshots
UQS
Warn Others

https://user-images.githubusercontent.com/72390277/184136502-42f25f63-df97-4a06-97c3-d102116d1fc2.MP4

### CODE REVIEW

File: src/xcode/ENA/ENA/Source/Scenes/QRScanner/QRScannerViewController.swift [hunk 1]
```diff
@@ -145,7 +148,7 @@ class QRScannerViewController: UIViewController {
 		instructionDescription.textAlignment = .center
 		instructionDescription.textColor = .enaColor(for: .iconWithText)
 		instructionDescription.font = .enaFont(for: .body)
-		instructionDescription.text = AppStrings.UniversalQRScanner.instructionDescription
+		instructionDescription.text = (presenter != nil && presenter == .onBehalfFlow) ? AppStrings.UniversalQRScanner.instructionDescriptionWarnOthers : AppStrings.UniversalQRScanner.instructionDescription
```
Comment (flxschmidt) [inline src/xcode/ENA/ENA/Source/Scenes/QRScanner/QRScannerViewController.swift:151]: Optional: to keep line good readable, you could use a private calc property like

```swift
instructionDescription.text = instructionDescription

...

private var instructionDescription: String {
   guard isPresenterOnBehalfFlow else { 
      return AppStrings.UniversalQRScanner.instructionDescription 
   }
   return AppStrings.UniversalQRScanner.instructionDescriptionWarnOthers
}
```

File: src/xcode/ENA/ENA/Source/Scenes/QRScanner/QRScannerViewController.swift [hunk 2]
```diff
@@ -173,6 +176,7 @@ class QRScannerViewController: UIViewController {
 		contentView.addSubview(instructionDescription)
 
 		infoButton.translatesAutoresizingMaskIntoConstraints = false
+		infoButton.isHidden = (presenter != nil && presenter == .onBehalfFlow)
```
Comment (flxschmidt) [inline src/xcode/ENA/ENA/Source/Scenes/QRScanner/QRScannerViewController.swift:179]: maybe also a logic that wraps that evaluation. 



```swift
infoButton.isHidden = isPresenterOnBehalfFlow

...

private var isPresenterOnBehalfFlow: Bool { presenter != nil && presenter == .onBehalfFlow }
```

**Review by flxschmidt — APPROVED** [review]
LGTM! Optional suggestions.

**Review by KathrinLuecke — APPROVED** [review]
UA approved

File: src/xcode/ENA/ENA/Source/Scenes/QRScanner/QRScannerViewController.swift (hunk 1 above)
Comment (puneetmahali1) [reply src/xcode/ENA/ENA/Source/Scenes/QRScanner/QRScannerViewController.swift:151]: Thanks Felix!
After discussed it with Naveed I did it in this way.

**Comment by Ein-Tim** [comment]
The info button is still there in version 2.27.0. Is this intended @puneetmahali1?

**Comment by Ein-Tim** [comment]
@puneetmahali1

Any news you can share with us?

**Pull request merged**